### PR TITLE
MRG: Update repository structure

### PR DESCRIPTION
**Pull request by mscheltienne**
As per #19, the structure could be something like that.

I chose `label` as name for the entry-point: adding one for the entire package (`mne_icalabel`) and one for each model (submodule, only `iclabel` for now). If you have a better name in mind, please comment 😉 
That would bring the public API to:

```
from mne_icalabel import label
label(raw, ica, method='iclabel')
```

And

```
from mne_icalabel.iclabel import label
label(raw, ica)
```

--------------------

I added an underscore to all private functions. I chose to keep the feature extraction `get_features`, the network `ICLabelNet`, the forward pass `run_iclabel` public on top of the entry-point `label`. 

WDYT?

### CODE REVIEW

**Comment by adam2392** [comment]
Thanks for getting to this right away!

I feel like `label` might be too generic and not explicit enough. Perhaps `label_ic`?

**Comment by mscheltienne** [comment]
How about `label_components` since MNE has the methods `get_components` and `plot_components` for ICA?

**Comment by adam2392** [comment]
> How about `label_components` since MNE has the methods `get_components` and `plot_components` for ICA?

I like it!

**Comment by mscheltienne** [comment]
Done, and I added the simplest of test cases to run `label_components` on the MNE raw sample dataset.. and it's already failing! 😅

**Comment by mscheltienne** [comment]
For the error, it's because autocorrelation ends with a resampling: 

```
# resample to 1 second at 100 samples/sec
resamp = resample_poly(ac.T, 100, np.round(raw.info["sfreq"])).T
```

The sample dataset sampling rate is `600.614990234375` Hz. It was complaining that the up-scaling and the down-scaling factors must be integers, so I rounded `raw.info['sfreq']`. But that's not the correct fix as now, instead of resampling to 100 samples, it resampled to 99.. thus failing the reshaping at a later stage.

I am not sure how to correctly fix this at the moment, I'll think about it.

**Comment by mscheltienne** [comment]
I agree that some error checking is necessary with some warning:
- Is the reference a common average? (we need more info than `'custom_ref_applied'` provide, there is an issue somewhere on the main repo about storing the reference for EEG dataset)
- Is the sampling frequency an integer? (not sure, but I think a float sfreq would crash in EEGLAB)
- Is the dataset BP between 0 and 100 Hz?
and some raises:
- Do we have the same channel in the ICA and in the instance? (ideally, you need to provide the instance used for fit)

We'll discuss those bullet points later, but for the frequency, I don't think we should force it to be integer because if a user has a dataset like the MNE sample, he can not easily change the sampling frequency. The field `.info['sfreq']` is among the lock fields that should not be tampered with and raises if you try to directly change it. Resampling doesn't seem like the way to go either as it will alter the data.. for no real reason.

---------------------

Instead, I would simply figure out if we have to floor or ceil the sampling frequency in the autocorrelation function to get the 100 samples. I am not (yet) convinced `floor` for every case is the correct method.

**Comment by adam2392** [comment]
> I agree that some error checking is necessary with some warning:

Agreed w/ all your warning proposals. Shall we merge this and go for it in another PR?

> We'll discuss those bullet points later, but for the frequency, I don't think we should force it to be integer because if a user has a dataset like the MNE sample, he can not easily change the sampling frequency. The field `.info['sfreq']` is among the lock fields that should not be tampered with and raises if you try to directly change it. Resampling doesn't seem like the way to go either as it will alter the data.. for no real reason.

Perhaps we have an optional argument `sfreq=None`, which is used to override the sampling rate. I think intelligently flooring/ceiling it is not possible since one could easily have `127.9584` or `128.12040` for example sampling rate, which actually corresponds to 128 Hz. WDYT of this soln? Con is adding an additional kwarg, but I don't see an easy way around w/o too much hacking.

File: mne_icalabel/iclabel/tests/test_label_components.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# preprocess
+raw.filter(l_freq=1.0, h_freq=100.0)
+raw.set_eeg_reference("average")
+# fit ICA
+ica = ICA(n_components=15, method="picard")
+ica.fit(raw)
+
+
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
+@pytest.mark.filterwarnings("ignore::FutureWarning")
```
Comment (adam2392) [inline mne_icalabel/iclabel/tests/test_label_components.py:21]: I will clean this up in the next PR.

**Review by adam2392 — APPROVED** [review]

File: mne_icalabel/iclabel/features.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 from numpy.typing import NDArray
 from scipy.signal import resample_poly
 
-from .utils import _next_power_of_2, gdatav4, mne_to_eeglab_locs, pol2cart
+from .utils import _gdatav4, _mne_to_eeglab_locs, _next_power_of_2, _pol2cart
 
 
 def get_features(inst: Union[BaseRaw, BaseEpochs], ica: ICA):
```
[the remark this answers is not in the document]
Comment (mscheltienne) [reply mne_icalabel/iclabel/features.py:13]: Yep, good idea.

File: mne_icalabel/iclabel/tests/test_label_components.py (hunk 1 above)
Comment (mscheltienne) [reply mne_icalabel/iclabel/tests/test_label_components.py:21]: Much appreciated!

**Comment by mscheltienne** [comment]
For the warnings, yes I will add those later on. 
For the optional ICA, sounds good to me! But we will have to document which settings we use by default. 

For the sampling rate, I think intelligent flooring/ceiling is possible, but your example is a very good point and I suspect it will fail with the current code. I am not that convince by the additional argument which complicates needlessly the API.
I will add a test case for the resampling alone, and hopefully a solution that will convince both of us that correct flooring/ceiling is possible :)

**Comment by adam2392** [comment]
Sounds good! FYI the repo is moved to my GitHub for now so I can work on the CI and docs before doing another migration to mne.tools :p 

You'll have to reset your remote URL

`git remote remove <current one>`
then
`git add remote <new one>`

**Comment by mscheltienne** [comment]
Perfect, don't worry about the move(s) ;)

**Pull request merged**